pkg/messaging/tracing/pubsub.go

```diff
@@ -0,0 +1,82 @@
+package tracing
+
+import (
+	"context"
+
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+const (
+	subscribeOP   = "subscribe_op"
+	unsubscribeOp = "unsubscribe_op"
+	handleOp      = "handle_op"
+)
+
+var _ messaging.PubSub = (*pubsubMiddleware)(nil)
+
+type pubsubMiddleware struct {
+	publisherMiddleware
+	pubsub messaging.PubSub
+	tracer opentracing.Tracer
+}
+
+// NewPubSub creates new pubsub tracing middleware service
+func NewPubSub(pubsub messaging.PubSub, tracer opentracing.Tracer) messaging.PubSub {
+	return &pubsubMiddleware{
+		publisherMiddleware: publisherMiddleware{
+			publisher: pubsub,
+			tracer:    tracer,
+		},
+		pubsub: pubsub,
+		tracer: tracer,
+	}
+}
+
+// Subscribe implements messaging.PubSub
+func (pm *pubsubMiddleware) Subscribe(ctx context.Context, id string, topic string, handler messaging.MessageHandler) error {
+	span, _ := opentracing.StartSpanFromContextWithTracer(ctx, pm.tracer, subscribeOP)
```

Comment:
```go
func (pm *pubsubMiddleware) createSpan(ctx context.Context, opName, topic, id string) opentracing.Span {
	if parentSpan := opentracing.SpanFromContext(ctx); parentSpan != nil {
		return pm.tracer.StartSpan(
			opName,
			opentracing.Tag{Key: "topic", Value: topic},
			opentracing.Tag{Key: "subscriber", Value: id},
			opentracing.ChildOf(parentSpan.Context()),
		)
	}
	return pm.tracer.StartSpan(
		opName,
		opentracing.Tag{Key: "topic", Value: topic},
		opentracing.Tag{Key: "subscriber", Value: id},
	)
}
```

Then use `createSpan`. You can modify it further to use in both `publisher.go` and `pubsub.go`